rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/NSCreationServlet.java

```diff
@@ -161,26 +163,48 @@ private void createNs(HttpServletRequest request, HttpServletResponse response)
+                                                                                               toArray(infraFileParams),
+                                                                                               policy,
+                                                                                               toArray(policyParams),
+                                                                                               toArray(policyFileParams),
+                                                                                               nodesRecoverable);
+                    break;
+                case "create":
+                    jsonResponsePayload = ((RMServiceImpl) RMServiceImpl.get()).defineNodeSource(sessionId,
+                                                                                                 nsName,
+                                                                                                 infra,
+                                                                                                 toArray(infraParams),
+                                                                                                 toArray(infraFileParams),
+                                                                                                 policy,
+                                                                                                 toArray(policyParams),
+                                                                                                 toArray(policyFileParams),
+                                                                                                 nodesRecoverable);
+                    break;
+                case "update":
+                    jsonResponsePayload = ((RMServiceImpl) RMServiceImpl.get()).updateDynamicParameters(sessionId,
+                                                                                                        nsName,
+                                                                                                        infra,
+                                                                                                        toArray(infraParams),
+                                                                                                        toArray(infraFileParams),
+                                                                                                        policy,
+                                                                                                        toArray(policyParams),
+                                                                                                        toArray(policyFileParams));
+                    break;
+                default:
+                    jsonResponsePayload = new JSONWriter(new StringWriter()).object()
+                                                                            .key("result")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "result" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)